Replacement of outmoded terminology and more accurate abstract+motivation (#2)
nucypher/mining-paper: mining-paper.tex

\begin{document}

\title{\nucypher: Staking Economics \& Stake Management}

\author{Michael Egorov, MacLane Wilkison}
\email{michael@nucypher.com, maclane@nucypher.com}
\affiliation{NuCypher}

\begin{abstract}
This paper describes mechanisms in the \nucypher~network designed to incentivize staking, including the regular distribution of an inflationary reward to active stakers, achieved via the growth of the native token's circulating supply. The paper explains the relationship between discretionary stake configurations and the staker's earning rate, in particular the extra incentive to commit to longer staking durations. Further rules for stake management, including the locking, unwinding and withdrawing of stake, are covered. The impact of these collective staker choices on the global supply dynamics is formally deduced. Finally, the paper provides an empirical rationale for the genesis inflation rate. 

\end{abstract}

\date{\today}
\maketitle

\section{Motivation}

> As adoption of the \nucypher~network increases, payments for access control services (`fees') will account for an increasing proportion of the earnings made by network nodes. However, before adoption has reached a sufficient threshold, nodes must be incentivized and subsidized via a regular inflationary reward.

Reviewer: ```suggestion
If adoption of the \nucypher~network increases, payments for access control services (`fees') may account for an increasing proportion of the earnings made by network nodes. However, before adoption has reached a sufficient threshold, nodes must be incentivized and subsidized via a regular inflationary reward.
```

This can be achieved through the scheduled growth of the native token's circulating supply, where newly minted tokens are distributed exclusively to active stakers.

A inflationary reward model of this sort should have the following properties:
\begin{itemize}
    \item The amount of rewards distributed to stakers is proportional to the relative size of their stake;
    \item The amount of work assigned to stakers (and, generally, the sum of fees earned) is proportional to the relative size of their stake;
    \item Stakers are incentivized to stake for longer durations by a commensurately higher reward rate;

>     \item Short-term staking is also encouraged, albeit at a lower reward rate;

Author: ```suggestion
    \item Short-term staking is acceptable, albeit at a lower reward rate;
```

    \item Epochs with higher levels of inflation do not significantly depreciate the native token price;

>     \item Stakers are incentivized to `re-stake' rewards in order to maximize their long-term compensation.

Reviewer: ```suggestion
    \item Stakers are incentivized to `re-stake' rewards in order to maximize their long-term stake in the network.
```

    \label{fig:mining-modes}
\end{figure}

At any point, a staker can split their stake or any piece of their stake into two pieces~(\figref{fig:mining-modes}).
The size of each piece should be not less than $S_{\min}$.

> The reason stakers may want to split their stake is because their compensation rate depends on the lock time $T$,

Reviewer: Agree that this is the underlying reason.

> which will be discussed in more details later.

Author: ```suggestion
which will be discussed in more detail later.
```

\subsection{Inflation decay}

> Initially, inflation subsidizes staking, but payments for re-encryption services will generate the majority or all of the revenues earned by stakers in the long run.

Reviewer: ```suggestion
Initially, inflation subsidizes staking, but payments for re-encryption services may generate the majority or all of the revenues earned by stakers in the long run as the inflation subsidy eventually runs out.
```

So, the amount of tokens distributed to the node $i$ and the time period $t$ will be:
\begin{eqnarray}
    \label{eq:rate-max}
    ds_{i,t} &=& \frac{l_i}{L} \frac{\ln{2}}{T_{1/2}} \left( S_{\max} - S_{t-1} \right),\\
    dS_t &=& \sum_i ds_{i,t},
\end{eqnarray}
where $l_i$ is the number of tokens locked by the staker $i$, $L$ is the total number of tokens locked.
Instead of calculating all the sum over $i$, each staker $i$ can add her portion $ds_{i,t}$.

\subsection{Reward rate and commitment time}

\begin{figure}
    \includegraphics[width=\columnwidth]{pdf/rate.pdf}
    \caption{Dependence of the compensation rate on staking duration. We assume $60\%$ of all tokens locked for staking}
    \label{fig:reward-rate-vs-duration}
\end{figure}

We want to incentivize stakers to serve re-encryption policies for at least $1$~year.
However, short-term stakers are still useful and should be compensated.
We will give the full compensation ($\kappa=1$) to the stakers who are committed to stake at least $T_1=1$~year,
however those who stake for $T_{\min}=1$~month will get close to half the compensation ($\kappa\approx0.54$) (\figref{fig:reward-rate-vs-duration}).

> The individual daily compensation rate for a staker looks as:

Reviewer 2: "looks as" reads weirdly.

```suggestion
The individual daily compensation rate for a staker is:
```

    S(t) = S_0 \left[1 + \frac{i_0 \kappa^* T_{1/2}^*}{\ln{2}}\left(1 - 2^{-\frac{t}{T_{1/2}^*}} \right) \right].
\end{equation}

\section{Staking strategies and expected compensation}

In this section, we look at three possibilities:
a staker liquidating all the compensation while extending the lock time (Stake and Earn),
a staker adding all the compensation to their current stake (Restake),

> and a staker waiting for their stake to unlock after time $T$ (Spindown).

Reviewer 2: Renamed to wind down:

```suggestion
and a staker waiting for their stake to unlock after time $T$ (Wind down).
```

    \label{fig:total-compensation}
\end{figure}

> \subsection{Take compensation and spindown}

Reviewer 2: ```suggestion
\subsection{Take compensation and wind down}
```

> When the node spins down, the staker doesn't extend the time for end of staking $t_1$,

Reviewer 2: ```suggestion
When the node winds down, the staker doesn't extend the time for end of staking $t_1$,
```

and the compensation is constantly decreasing as the time left to unlock becomes smaller and smaller,
effectively decreasing $\kappa$ gradually towards $0.5$.

> That's the default behavior: to avoid that, the staker should set $t_1$ large enough, or increase $t_1$ periodically.

Reviewer 2: This is no longer the default...The default is that wind down is disabled.

Author: ```suggestion
The default setting for all stakes is for $t_1$ to be automatically prolonged, which maintains the maximum reward rate ($\kappa$ = 1). Stakers wishing to withdraw must proactively initiate a wind down on some or all of their stake.
```

\subsection{Edge case: connection problems}

If a staker is found to be non-operational and/or cannot confirm its continued performance, the tokens aren't unlocked (and compensation isn't earned).
It's not fatal for the staker if that happens (i.e. their entire stake won't be slashed), however their tokens will remain locked without earning compensation,
until they satisfy their commitment.
So if the staker commits to stake for at least a year, it implies a year of operation.

> Connection problems may also be downtime when the staker upgrades their node's software, which is an absolutely legitimate reason to be offline for a short time.

Reviewer 2: ```suggestion
Connection problems may include downtime when the staker upgrades their node's software, which is a legitimate reason to be offline for a short time.
```

But if you restake all your tokens, in the same first year you'd earn $177.5\%$ of the tokens you staked.

\subsection{How many tokens will ever be in existence?}
We'll start with $1$~billion tokens, and the maximum amount of tokens ever minted and distributed will be $3.89$~billion.

\subsection{What's the inflation rate?}
The inflation rate will depend on how many short-term stakers and long-term stakers are working in the system.

> Depending on this, the initial inflation will be between $50\%$~APR (if all stakers are very short term) and $100\%$~APR (if all stakers commit for a long

Reviewer 2: For this statement and the next, is it a question of **all stakers are...** or **a sufficient number/majority of stakers are...**?
